plugins/tiledb/io/TileDBReader.cpp

```diff
@@ -0,0 +1,298 @@
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+* FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+* COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+* BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+* OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+* OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+* OF SUCH DAMAGE.
+****************************************************************************/
+
+#include <algorithm>
+
+#include <boost/any.hpp>
+
+#include "TileDBReader.hpp"
+
+namespace pdal {
+
+static PluginInfo const s_info
+{
+        "readers.tiledb",
+        "TileDB Reader",
+        "http://pdal.io/stages/readers.tiledb.html"
+};
+
+CREATE_SHARED_STAGE(TileDBReader, s_info)
+std::string TileDBReader::getName() const { return s_info.name; }
+
+template <typename T>
+void makeBuffer(tiledb::Query& query, std::map<std::string, pdalboost::any>& mp,  const std::string& key,
+        const unsigned bufSize)
+{
+    auto sp = std::make_shared<std::vector<T>>(bufSize);
+    mp[key] = sp;
+
+    if (key != TILEDB_COORDS)
+        query.set_buffer(key, *sp);
+}
+
+void TileDBReader::addArgs(ProgramArgs& args)
+{
+    args.add("array_name", "TileDB array name", m_arrayName).setPositional();
+    args.add("config_file", "TileDB configuration file location", m_cfgFileName);
+    args.add("chunk_size", "TileDB read chunk size", m_chunkSize, point_count_t(1000));
+}
+
+void TileDBReader::initialize()
+{
+    if (!m_cfgFileName.empty())
+    {
+        tiledb::Config cfg(m_cfgFileName);
+        m_ctx.reset(new tiledb::Context(cfg));
+    }
+    else
+        m_ctx.reset(new tiledb::Context());
+
+    m_array.reset(new tiledb::Array(*m_ctx, m_arrayName, TILEDB_READ));
+}
+
+void TileDBReader::addDimensions(PointLayoutPtr layout)
+{
+    std::vector<tiledb::Dimension> dims = m_array->schema().domain().dimensions();
+    auto attrs = m_array->schema().attributes();
+
+    for (const auto& dim: dims)
+    {
+        layout->registerDim(Dimension::id(dim.name()));
+        m_ranges.push_back(dim.domain<double>().first);
+        m_ranges.push_back(dim.domain<double>().second);
+    }
+
+
+    for (const auto& a : attrs) {
+        std::string attrName = a.first;
+        Dimension::Id id = Dimension::id(attrName);
+        layout->registerDim(id);
+    }
+
+}
+
+point_count_t TileDBReader::read(PointViewPtr view, point_count_t count)
+{
+    PointId idx = view->size();
+    PointRef point(*view, idx);
+    point_count_t numRead = 0;
+    tiledb::Query query(*m_ctx, *m_array);
```

Comment:
I didn't see anywhere here where a subarray is being set on the query -- should there be one set?

Reply:
Not in this case, the reader doesn't subset.